cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -717,7 +719,13 @@ public MapAndStatus(T loadMap, K status) {
             return onNew(reconciliation, host, apiClient, kafkaRebalance, rebalanceOptionsBuilder);
         } else {
             // Stay in the current NotReady state, returning null as next state
-            return Future.succeededFuture(new MapAndStatus<>(null, null));
+            if (rawRebalanceAnnotation(kafkaRebalance) == null) {
```

Comment:
if raw annonation is `null`, it means to be `none` as enum, why are we anyway doing this check here? 
Anyway the code can be simpler. You are anyway calling `validate` to get the conditions but then adding the invalid annotation condition only if it's not `none` (so any other different from the valid `refresh`).
So in the `else` branch, you could have just something like this? 

```java
Set<Condition> conditions = validate(reconciliation, kafkaRebalance);
            if (rebalanceAnnotation != KafkaRebalanceAnnotation.none) {
                conditions.add(StatusUtils.buildWarningCondition("InvalidAnnotation", "Wrong annotation value: " + rawRebalanceAnnotation(kafkaRebalance) + ". The rebalance resource supports the following annotations in `NotReady` state: refresh"));
            }
            return Future.succeededFuture(buildRebalanceStatus(null, KafkaRebalanceState.NotReady, conditions));
```

Reply:
While applying the suggestion, One edge case slipped into my brain. What if we are on `Ready` state and a user passes `none` annotation? It will make the condition to be false and we will not get the warning

Reply:
Should we treat `none` as `null` only then ? ie if someone passes `none`, we don't do anything and dont add the warning either?

Reply:
`none` is not an annotation value :-) ... `none` is the enum val corresponding to the fact that there is NO annotation on the `KafkaRebalance` resource (so raw annotation is `null` because getting annotation from the custom resource return `null` because it doesn't exist).